lib/writers.js

```diff
@@ -194,63 +202,56 @@ class WriteQueue extends events.EventEmitter {
 
   run() {
     if (!this.isRunning) {
-      this.process();
+      this.isRunning = true;
+      // Use nextTick to allow the queue to build up on the current phase
```

Comment:
This comment was really useful in helping me understand why `nextTick`.  We want to allow executing code to finish to allow queue to build up, instead of invoking right away each time.  Seems like could be introduce a small improvement.